modules/aggregation-layers/src/cpu-grid-layer/cpu-grid-layer.js

```diff
@@ -181,15 +178,31 @@ export default class CPUGridLayer extends CompositeLayer {
     };
   }
 
-  getDimensionChanges(oldProps, props) {
+  needUpdateDimensionStep(dimensionStep, oldProps, props, changeFlags) {
+    // whether need to update current dimension step
+    // dimension step is the value, domain, scaleFunction of each dimension
+    return dimensionStep.triggers.some(t => {
+      if (dimensionStep.updateTriggers && dimensionStep.updateTriggers[t]) {
+        // check based on updateTriggers change first
+        return (
+          changeFlags.updateTriggersChanged &&
+          (changeFlags.updateTriggersChanged.all || changeFlags.updateTriggersChanged[t])
+        );
+      }
+      // fallback to direct comparison
+      return oldProps[t] !== props[t];
```

Comment:
This is inconsistent with the rest of the layers - we ignore shallow change in all accessors and exclusively rely on update triggers.

Reply:
direct comparison only applies to layer props that are not an accessor. For all accessors in dimensionUpdater, we only check updateTriggers